src/mlpack/methods/cf/cf.hpp

```diff
@@ -73,10 +56,11 @@ struct FactorizerTraits
  * are in a matrix that holds doubles, should hold integer (or size_t) values.
  * The user and item indices are assumed to start at 0.
  *
- * @tparam FactorizerType The type of matrix factorization to use to decompose
+ * @tparam DecompositionPolicy The algorithm to use to decompose
  *     the rating matrix (a W and H matrix).  This must implement the method
  *     Apply(arma::sp_mat& data, size_t rank, arma::mat& W, arma::mat& H).
  */
+template<typename DecompositionPolicy = NMFPolicy>
 class CF
```

Comment:
To allow backwards compatibility, we should call this CFType, and provide an alias that uses, ``amf::NMFALSFactorizer``. Something like:

```c++
using CF = CFType<amf::NMFALSFactorizer>;
```

that way, the old interface remains since I can just reuse the ``CF`` class without the template parameter.